src/ext/conversion/import-properties.ts

```diff
@@ -0,0 +1,89 @@
+import conversion from 'qlik-object-conversion';
+import { setValue } from 'qlik-chart-modules';
+
+import data from '../../qae/data';
+import { ExportFormat, PropTree, DimensionProperties, MeasureProperties } from '../../types';
+import { ColumnWidthTypes } from '../../table/constants';
+
+export const getColumnInfo = (
+  columnInfo: DimensionProperties[] | MeasureProperties[],
+  colIdx: number,
+  columnWidths?: number[],
+  numDims?: number
+) => {
+  let index = colIdx;
+  if (numDims) index = colIdx - numDims;
+  const column = JSON.parse(JSON.stringify(columnInfo[index]));
```

Comment:
why not { ...columnInfo[index] } ?

Reply:
{ ...columnInfo[index] } is for shallow copy, and we need deep copy here
https://makimo.pl/blog/shallow-and-deep-copies-in-javascript/#:~:text=We%20have%20two%20kinds%20of,that%20contain%20only%20primitive%20values.&text=Nested%20objects%20mean%20objects%20that%20contain%20non%2Dprimitive%20values.

Reply:
huh, I thought spreading was doing a deep copy. Really good to know. Then this is fine